File: software/gitswamp/src/types/models/stagedDiffSummary.ts
export interface StagedDiffSummary {
  total_lines_added: number;
  total_lines_removed: number;
  files_changed: number;
  file_types: string[];
  has_test_changes: boolean;
  has_migration_changes: boolean;
  inferred_scope: string;
}

File: software/gitswamp/src-tauri/src/models/staged_diff_summary.rs
use serde::Serialize;

#[derive(Debug, Clone, Serialize)]
pub struct StagedDiffSummary {
    pub total_lines_added: usize,
    pub total_lines_removed: usize,
    pub files_changed: usize,
    pub file_types: Vec<String>,
    pub has_test_changes: bool,
    pub has_migration_changes: bool,
    pub inferred_scope: String,
}
